Mono's GC fails on sigaltstack w/ SS_DISABLE

Repro steps

```
Bin/ELFLoader -s -c irjit -n 500 -- /usr/bin/mono /usr/lib/mono/4.5/mcs.exe
error CS2008: No files to compile were specified
Compilation failed: 1 error(s), 0 warnings
Mono: GC_MAJOR: (user request) time 50.11ms, stw 77.87ms los size: 0K in use: 0K
Mono: GC_MAJOR_SWEEP: major size: 768K in use: 81K
* Assertion at mini-exceptions.c:3191, condition `err == 0' not met
```

Mono Source: https://github.com/mono/mono/blob/master/mono/mini/mini-exceptions.c#L3196
```
mono_free_altstack (MonoJitTlsData *tls)
{
    stack_t sa;
    int err;

    sa.ss_sp = tls->signal_stack;
    sa.ss_size = MONO_ARCH_SIGNAL_STACK_SIZE;
    sa.ss_flags = SS_DISABLE;
    err = sigaltstack  (&sa, NULL);
    g_assert (err == 0);
```

--- a/External/FEXCore/Source/Interface/Core/Interpreter/Arm64Dispatcher.cpp
+++ b/External/FEXCore/Source/Interface/Core/Interpreter/Arm64Dispatcher.cpp
@@ -425,7 +425,7 @@ bool DispatchGenerator::HandleGuestSignal(int Signal, void *info, void *ucontext
   uint64_t OldGuestSP = State->State.State.gregs[X86State::REG_RSP];
   uint64_t NewGuestSP = OldGuestSP;
 
-  if (!!GuestStack->ss_sp) {
+  if (!(GuestStack->ss_flags & SS_DISABLE)) {
     // If our guest is already inside of the alternative stack
     // Then that means we are hitting recursive signals and we need to walk back the stack correctly
     uint64_t AltStackBase = reinterpret_cast<uint64_t>(GuestStack->ss_sp);

--- a/External/FEXCore/Source/Interface/Core/Interpreter/x86_64Dispatcher.cpp
+++ b/External/FEXCore/Source/Interface/Core/Interpreter/x86_64Dispatcher.cpp
@@ -296,7 +296,7 @@ bool DispatchGenerator::HandleGuestSignal(int Signal, void *info, void *ucontext
   uint64_t OldGuestSP = State->State.State.gregs[X86State::REG_RSP];
   uint64_t NewGuestSP = OldGuestSP;
 
-  if (!!GuestStack->ss_sp) {
+  if (!(GuestStack->ss_flags & SS_DISABLE)) {
     // If our guest is already inside of the alternative stack
     // Then that means we are hitting recursive signals and we need to walk back the stack correctly
     uint64_t AltStackBase = reinterpret_cast<uint64_t>(GuestStack->ss_sp);

--- a/External/FEXCore/Source/Interface/Core/JIT/Arm64/JIT.cpp
+++ b/External/FEXCore/Source/Interface/Core/JIT/Arm64/JIT.cpp
@@ -211,7 +211,7 @@ bool JITCore::HandleGuestSignal(int Signal, void *info, void *ucontext, SignalDe
   uint64_t OldGuestSP = State->State.State.gregs[X86State::REG_RSP];
   uint64_t NewGuestSP = OldGuestSP;
 
-  if (!!GuestStack->ss_sp) {
+  if (!(GuestStack->ss_flags & SS_DISABLE)) {
     // If our guest is already inside of the alternative stack
     // Then that means we are hitting recursive signals and we need to walk back the stack correctly
     uint64_t AltStackBase = reinterpret_cast<uint64_t>(GuestStack->ss_sp);

--- a/External/FEXCore/Source/Interface/Core/JIT/x86_64/JIT.cpp
+++ b/External/FEXCore/Source/Interface/Core/JIT/x86_64/JIT.cpp
@@ -135,7 +135,7 @@ bool JITCore::HandleGuestSignal(int Signal, void *info, void *ucontext, SignalDe
   uint64_t OldGuestSP = ThreadState->State.State.gregs[X86State::REG_RSP];
   uint64_t NewGuestSP = OldGuestSP;
 
-  if (!!GuestStack->ss_sp) {
+  if (!(GuestStack->ss_flags & SS_DISABLE)) {
     // If our guest is already inside of the alternative stack
     // Then that means we are hitting recursive signals and we need to walk back the stack correctly
     uint64_t AltStackBase = reinterpret_cast<uint64_t>(GuestStack->ss_sp);

--- a/External/FEXCore/Source/Interface/Core/SignalDelegator.cpp
+++ b/External/FEXCore/Source/Interface/Core/SignalDelegator.cpp
@@ -17,7 +17,11 @@ namespace FEXCore {
   struct ThreadState {
     FEXCore::Core::InternalThreadState *Thread{};
     void *AltStackPtr{};
-    stack_t GuestAltStack{};
+    stack_t GuestAltStack {
+      .ss_sp = nullptr,
+      .ss_flags = SS_DISABLE, // By default the guest alt stack is disabled
+      .ss_size = 0,
+    };
     // Guest signal sa_mask is per thread!
     SignalDelegator::GuestSAMask Guest_sa_mask[SignalDelegator::MAX_SIGNALS]{};
     uint32_t CurrentSignal{};
@@ -362,7 +366,8 @@ namespace FEXCore {
     uint64_t AltStackEnd = AltStackBase + ThreadData.GuestAltStack.ss_size;
     uint64_t GuestSP = ThreadData.Thread->State.State.gregs[X86State::REG_RSP];
 
-    if (GuestSP >= AltStackBase &&
+    if (!(ThreadData.GuestAltStack.ss_flags & SS_DISABLE) &&
+        GuestSP >= AltStackBase &&
         GuestSP <= AltStackEnd) {
       UsingAltStack = true;
     }
@@ -389,10 +394,16 @@ namespace FEXCore {
       }
 
       // We need to check for invalid flags
-      // The only flag that can be passed is SS_AUTODISARM
-      if (ss->ss_flags & ~SS_AUTODISARM) {
+      // The only flag that can be passed is SS_AUTODISARM and SS_DISABLE
+      if (ss->ss_flags & ~(SS_AUTODISARM | SS_DISABLE)) {
         // A flag remained that isn't one of the supported ones?
         return -EINVAL;
+      }
+
+      if (ss->ss_flags & SS_DISABLE) {
+        // If SS_DISABLE Is specified then the rest of the details are ignored
+        ThreadData.GuestAltStack = *ss;
+        return 0;
       }
 
       // stack size needs to be MINSIGSTKSZ (0x2000)

